### ISSUE TITLE
Automatically set mermaid themes to match dark / light

### ISSUE THREAD

**Issue by cscheid** [quarto-dev/quarto-cli#1530]
### Discussed in https://github.com/quarto-dev/quarto-cli/discussions/1521

<div type='discussions-op-text'>

<sup>Originally posted by **edavidaja** July 25, 2022</sup>
currently I'm using a reveal presentation with `theme: night` and adding 

`%%{init: {"theme": "dark" } }%%`
 
to each diagram. I'm not sure if this works for every diagram type--the mermaid website suggests that theming, for example, state machines has to be done via a different mechanism.</div>

**Comment by mcanouil** [comment]
cc @gordonwoodhull

**Comment by gordonwoodhull** [comment]
Yeah I'm not doing this for 1.7 but I'd like to revisit both Mermaid and Graphviz light and dark themes soon.
